pkg/kubernetes/kubernetes_interface.gen.go

```diff
@@ -9,157 +9,232 @@ import (
+	// ListDatabaseClusterBackups returns database cluster backups.
+	// This method returns a list of full objects (meta and spec).
+	ListDatabaseClusterBackups(ctx context.Context, opts ...ctrlclient.ListOption) (*everestv1alpha1.DatabaseClusterBackupList, error)
+	// UpdateDatabaseClusterBackup updates database cluster backup.
+	UpdateDatabaseClusterBackup(ctx context.Context, backup *everestv1alpha1.DatabaseClusterBackup) (*everestv1alpha1.DatabaseClusterBackup, error)
+	// DeleteDatabaseClusterBackup deletes database cluster backup.
+	DeleteDatabaseClusterBackup(ctx context.Context, obj *everestv1alpha1.DatabaseClusterBackup) error
+	// CreateDatabaseClusterBackup creates database cluster backup.
+	CreateDatabaseClusterBackup(ctx context.Context, backup *everestv1alpha1.DatabaseClusterBackup) (*everestv1alpha1.DatabaseClusterBackup, error)
+	// GetDatabaseClusterRestore returns database cluster restore by name.
+	GetDatabaseClusterRestore(ctx context.Context, key ctrlclient.ObjectKey) (*everestv1alpha1.DatabaseClusterRestore, error)
+	// ListDatabaseClusterRestores returns database cluster restores.
+	// This method returns a list of full objects (meta and spec).
+	ListDatabaseClusterRestores(ctx context.Context, opts ...ctrlclient.ListOption) (*everestv1alpha1.DatabaseClusterRestoreList, error)
+	// UpdateDatabaseClusterRestore updates database cluster restore.
+	UpdateDatabaseClusterRestore(ctx context.Context, restore *everestv1alpha1.DatabaseClusterRestore) (*everestv1alpha1.DatabaseClusterRestore, error)
+	// DeleteDatabaseClusterRestore deletes database cluster restore.
+	DeleteDatabaseClusterRestore(ctx context.Context, obj *everestv1alpha1.DatabaseClusterRestore) error
+	// CreateDatabaseClusterRestore creates database cluster restore.
+	CreateDatabaseClusterRestore(ctx context.Context, restore *everestv1alpha1.DatabaseClusterRestore) (*everestv1alpha1.DatabaseClusterRestore, error)
 	// ListDatabaseEngines returns list of managed database clusters.
-	ListDatabaseEngines(ctx context.Context, namespace string) (*everestv1alpha1.DatabaseEngineList, error)
+	// This method returns a list of full objects (meta and spec).
+	ListDatabaseEngines(ctx context.Context, opts ...ctrlclient.ListOption) (*everestv1alpha1.DatabaseEngineList, error)
 	// GetDatabaseEngine returns database clusters by provided name.
-	GetDatabaseEngine(ctx context.Context, namespace, name string) (*everestv1alpha1.DatabaseEngine, error)
+	GetDatabaseEngine(ctx context.Context, key ctrlclient.ObjectKey) (*everestv1alpha1.DatabaseEngine, error)
 	// UpdateDatabaseEngine updates the provided database engine.
 	UpdateDatabaseEngine(ctx context.Context, engine *everestv1alpha1.DatabaseEngine) (*everestv1alpha1.DatabaseEngine, error)
 	// SetDatabaseEngineLock sets the lock on the database engine.
 	// The lock is automatically set to false once everest-operator completes its upgrade.
 	SetDatabaseEngineLock(ctx context.Context, namespace, name string, locked bool) error
```

Comment:
Should we use `ctrlclient.Objectkey` here as well?

Reply:
em, interesting, because in my local code `ctrlclient.Objectkey` is used.